index.js

```diff
@@ -0,0 +1,135 @@
+const exec = require('@actions/exec');
+const core = require('@actions/core');
+const github = require('@actions/github');
+const { parse: gitDiffParser } = require('what-the-diff');
+const fs = require('fs-extra');
+const simpleGit = require('simple-git');
+const env = process.env;
+
+function matchExact(r, str) {
+  let match = str.match(r);
+  return match && str === match[0];
+}
+
+function fixNewLineEOF(b) {
+  // If file is empty
+  if (b.length < 1) {
+    return b;
+  }
+  // If file is not empty
+  b = b.replace(/[ \t\n]*$/, '\n');
+
+  if (b.length === 1) {
+    // if the remaining character is a whitespace
+    if (/[ \t\n]*$/.test(b)) {
+      return '';
+    } else {
+      return b + '\n';
+    }
+  }
+  return b;
+}
+
+async function getChangedFilesPaths(pull_request, octokit, owner, repo) {
+  const { data: pullRequestDiff } = await octokit.pulls.get({
+    owner,
+    repo,
+    pull_number: pull_request.number,
+    mediaType: {
+      format: 'diff'
+    }
+  });
+
+  const parsedDiff = gitDiffParser(pullRequestDiff);
+
+  const changedFilePaths = parsedDiff.map((e) => {
+    return e['newPath'].replace('b/', '');
+  });
+
+  return changedFilePaths;
+}
+
+async function run() {
+  const token = core.getInput('GH_TOKEN');
+  let ignorePaths = core.getInput('IGNORE_PATHS');
```

Comment:
The paths received must be in standard javascript regex.
Example:
An input of `dist/` does not ignore all the files inside the dist folder. Instead `dist/.*` needed to be used.
There are alternatives to support `dist/` but then all the possible cases must be manually defined.
One such case is:

```javascript
ignorePaths = ignorePaths.map((e) => {
	if (e.slice(-1) === '/') {
		return e + '.*';
	} else {
		return e;
	}
});
```

Reply:
JS Regex is fine. No need to do anything further.